articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,161 @@
+ Refer [this article](https://www.section.io/engineering-education/what-are-reverse-proxies/) to better understand what Reverse Proxies are.
+
+### NGINX
+We will be using NGINX which would be serving the purpose of a Reverse Proxy. NGINX is a web server that can be used as a reverse proxy, load balancer, mail proxy and HTTP cache. The software was created by Igor Sysoev and publicly released in 2004. Nginx is free and open-source software, released under the terms of the 2-clause BSD license. A large fraction of web servers use NGINX, often as a load balancer.
+
+To know more about NGINX, check out the [Wiki](https://en.wikipedia.org/wiki/Nginx).
+
+Some of the other Reverse Proxies available are:
+- [Apache](https://httpd.apache.org/docs/2.4/howto/reverse_proxy.html)
+- [Varnish](https://varnish-cache.org/intro/)
+
+
+### Reverse Proxy Example
+![image](./reverse_proxy_working.png)
+
+This is an example of an architecture, where two apps are running in the background, but clients have no idea about them. Clients only know about NGINX which acts as a reverse proxy that sends the request to the appropriate application.
+
+
+Now that you have a broad idea of what we are about to build, let us jump right in!
+
+## Aim
+1. Deploy two applications and have it managed by NGINX.
+
+## Setup & Pre-Requisites:
+
+1. For this example, I have two sample Express Applications. One can have any kind of application running on different ports. 
+**NOTE:** Do not run your application on Port 80 or 443. Will be explained later why this must not be done.  
+Refer the official ExpressJS Documentation for Getting Started. [Docs](https://expressjs.com/en/starter/installing.html)
+2. I have installed NGINX on my local machine, the same could be done on any Virtual Machine where the applications are expected to be deployed.  
+Here is the Documentatino on how to install NGINX on your machine. [Docs](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/) . 
+
+## Step 1: Start two apps running in different ports
+As I've mentioned earlier, I got two Node Apps running on two different ports as shown below.  
+
+**Server app running on Port 3000**
+
+![image](./server1.png)  
+
+**Client app running on Port 3001**  
+
+![image](./client1.png)
+
+Now that we have our apps up and running, we don't want our users to use these applications by typing their PORTS explicitly, so we need to map it with something which is more human-readable. In this example, I will be using subdomains to distinguish between them. Again one is free to use whichever distinguishing element suitable as per their requirement.
+Other example could be  a particular route like domain/client and domain/server. The only condition for the distinguishing element is to follow a Valid URL Regular Expression. Learn about Regex from [here](https://regexr.com/).
```

Comment:
**Another** example could be **(remove extra space** a particular route like domain/client **or** domain/server

Reply:
Thanks for pointing it out